autopush/endpoint.py

```diff
@@ -254,6 +271,41 @@ def _db_error_handling(self, d):
         d.addErrback(self._response_err)
         return d
 
+    def _process_auth(self, result):
+        crypto_key = None
+        header_info = parse_header(self.request.headers.get('crypto-key'))
+        if header_info:
+            # value dictionary is always the last item in the list
+            values = header_info[-1]
+            if isinstance(values, dict):
+                crypto_key = values.get('p256ecdsa')
+            else:
+                raise VapidAuthException("Invalid bearer token: "
+                                         "improperly specified crypto-key")
+        authorization = self.request.headers.get('authorization')
+        if crypto_key and authorization:
+            (auth_type, token) = authorization.split(' ', 1)
+            # if it's a bearer token containing what may be a JWT
+            if auth_type.lower() == 'bearer' and '.' in token:
+                try:
+                    jwt = extract_jwt(token, crypto_key)
+                    if jwt.get('exp') and jwt.get('exp') < time.time():
```

Comment:
Probably better to just get exp out on its own, so all the extra get() calls aren't needed (like the one below at 295).